File: backend/bookingmovieticketapp/src/main/java/com/project/bookingmovieticketapp/DTOs/UserDTO.java
package com.project.bookingmovieticketapp.DTOs;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.util.Date;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class UserDTO {
    @NotBlank(message = "Họ và tên không được bỏ trống")
    private String name;

    @NotBlank(message = "Email không được bỏ trống")
    private String email;

    @NotBlank(message = "Password không được bỏ trống")
    private String password;

    @NotBlank(message = "Xác nhận mật khẩu không khớp")
    private String retypepassword;

    @NotBlank(message = "Số điện thoại không được bỏ trống")
    private String phonenumber;

    @NotNull(message = "Ngày sinh không được bỏ trống")
    @JsonFormat(pattern = "yyyy-MM-dd")
    private Date dateofbirth;
}

File: frontendapp/lib/dtos/UserDTO.dart
class UserDTO{
  final String name;
  final String email;
  final String password;
  final String retypepassword;
  final String phonenumber;
  final String dateofbirth;

  UserDTO({
    required this.name,
    required this.email,
    required this.password,
    required this.retypepassword,
    required this.phonenumber,
    required this.dateofbirth,
  });

  Map<String,dynamic> toJson() {
    return {
      'name': name,
      'email': email,
      'password': password,
      'retypepassword': retypepassword,
      'phonenumber': phonenumber,
      'dateofbirth': dateofbirth,
    };
  }
}